src/plasma/plasma_manager.cc

```diff
@@ -235,12 +235,11 @@ struct PlasmaManagerState {
   /** The time (in milliseconds since the Unix epoch) when the most recent
    *  heartbeat was sent. */
   int64_t previous_heartbeat_time;
-  /** An object's object_id is added to this set immediately after
-   *  it is received and sealed. The object is removed in
-   *  process_add_object_notification, which is triggered by
-   *  the corresponding notification from the plasma store. */
-  // TODO(hme): change semantics to "received some bytes of object"
-  std::unordered_set<ObjectID, UniqueIDHasher> received_objects;
+  /** An ObjectID is added to this set if a shared buffer is
+   *  successfully created for the corresponding object.
+   *  The ObjectID is removed in process_add_object_notification, which is
+   *  triggered by the corresponding notification from the plasma store. */
+  std::unordered_set<ObjectID, UniqueIDHasher> receives_in_progress;
```

Comment:
Are we handling the case where a remote manager dies? In that case, we should flush the entire set.

Reply:
@elibol erases the entry when the object is aborted. Assuming we keep the current design where we only receive one object at a time from each remote manager, I don't think it's necessary to flush the entire set.

Reply:
I see, I missed that. I was thinking that we might implement something like #1485 which would introduce a bug. It wouldn't cause any test failures and it would only happen rarely.

I'm ok with either case, but if we leave it the way it is we should document very clearly the assumption being made about only one object being transferred at a time from a given manager.

I think flushing the whole set is ok because 1) the set should always be pretty small anyway and 2) reissuing a transfer request accidentally is not the end of the world.